ph5/clients/ph5toms.py

```diff
@@ -148,45 +149,100 @@ def read_events(self, name):
     def filenamemseed_gen(self, stream):
 
         s = stream.traces[0].stats
+        new_start = s.starttime.isoformat()
+        try:
+            rounded = ph5utils.roundSeconds(
+                datetime.datetime.strptime(
+                    new_start,
+                    "%Y-%m-%dT%H:%M:%S.%f"))
+        except BaseException:
+            rounded = ph5utils.roundSeconds(
+                datetime.datetime.strptime(
+                    new_start,
+                    "%Y-%m-%dT%H:%M:%S"))
         ret = "{0}.{1}.{2}.{3}.{4}.ms".format(
             s.network, s.station, s.location,
-            s.channel, s.starttime.strftime("%Y-%m-%dT%H.%M.%S"))
+            s.channel, rounded.strftime("%Y-%m-%dT%H.%M.%S"))
         if not self.stream:
             ret = os.path.join(self.out_dir, ret)
         return ret
 
     def filenamesac_gen(self, trace):
         s = trace.stats
+        new_start = s.starttime.isoformat()
+        try:
+            rounded = ph5utils.roundSeconds(
+                datetime.datetime.strptime(
+                    new_start,
+                    "%Y-%m-%dT%H:%M:%S.%f"))
+        except BaseException:
+            rounded = ph5utils.roundSeconds(
+                datetime.datetime.strptime(
+                    new_start,
+                    "%Y-%m-%dT%H:%M:%S"))

```

Comment:
Why did you choose to round to the nearest second? I use these methods for naming  the SAC files returned by the web service as well.

Reply:
I've had 5 PIs all say the same thing.  Having the microseconds in the names make it hard to read and can the file names quickly. Especially when it comes to requesting large amounts of data cut into days. This really only is a problem with the nodes  and their issues not starting to record on an even second. Since all the nodes start at different times and not on the even seconds, some will cut exactly on the day boundaries, some just before and some just after making the file names messy, when really all the PI wants is to know this file is day x of station y. They don't really care that it started a sample into the previous day or a sample into the new day